webui/src/lib/onboarding/repoOnboardingService.ts

```diff
@@ -0,0 +1,119 @@
+const HOOKS_PATH = "_lakefs_actions/";
+const SAMPLE_REPO_FILE_NAME = "lakes.parquet";
+
+// generate a random file name to avoid collisions for the upload preflight check
+const generateRandomFileName = (): string => {
+  const randomString = Math.random().toString(36).substring(2, 15);
+  return `sample_${randomString}.yaml`;
+};
+
+export const isSampleRepo = async (
+  repo: string,
+  defaultBranch = "main"
+): Promise<boolean> => {
+  const objectList = await objects.list(repo, defaultBranch, "");
+  return objectList.results.some(
+    // when the API client is typed, we can remove the eslint-disable
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (object: any) => object?.path === SAMPLE_REPO_FILE_NAME
+  );
+};
+
+export const canUseRepoOnboarding = async (
+  repoId: string,
+  defaultBranch = "main"
+): Promise<boolean> => {
+  const promises = [];
+  promises.push(retention.setGCPolicyPreflight(repoId));
+  promises.push(branchProtectionRules.createRulePreflight(repoId));
+  promises.push(
+    objects.list(
+      repoId,
+      defaultBranch,
+      `${HOOKS_PATH}${generateRandomFileName()}`
+    )
+  );
+  const results = await Promise.allSettled(promises);
+  if (results.some((result) => result.status === "rejected" || !result.value)) {
+    return false;
+  }
+  return true;
+};
+
+export const getRepoOnboardingSteps = (
+  currentRepo: string,
+  defaultBranch: string,
+  objectStoreName: string,
+  navigate: NavigateFunction
+): OnBoardingSteps => [
+  {
+    id: "import-data",
+    title: "Import your existing data",
+    description: `Import your existing data from ${objectStoreName} to your repository by creating pointers to it (no copying of data involved)`,
+    cta: "Run import",
+    onClick: () =>
+      navigate(`/repositories/${currentRepo}/objects?importDialog=true`),
+    showStep: () => objectStoreName !== "local",
+    isCompleted: async () => {
+      const repoBranches = await branches.list(currentRepo);
+      // when the API client is typed, we can remove the eslint-disable
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return repoBranches.results.some((branch: any) =>
+        IMPORT_BRANCH_REGEX.test(branch.id)
+      );
+    },
+  },
+  {
+    id: "garbage-collection",
+    title: "Setup garbage collection",
+    description:
+      "Setup rules to automatically manage when data can be deleted from your repository",
+    cta: "Configure",
+    onClick: () => navigate(`/repositories/${currentRepo}/settings/retention`),
+    showStep: () => true,
+    isCompleted: async () => {
+      try {
+        await retention.getGCPolicy(currentRepo);
+        return true;
+      } catch (e) {
+        return false;
+      }
+    },
+  },
+  {
+    id: "branch-protection",
+    title: "Set up branch protection",
+    description:
+      "Add a quality gate to ensure your most important branches pass quality, compliance, or privacy checks",
```

Comment:
This text seems misleading.  A branch protection rule does not add a "quality gate to ensure your most important branches pass quality, compliance, or privacy checks".  It just means that "changes [...] can only be done by merging from other branches."

Reply:
@ozkatz I think @arielshaqed makes a good point. The text is from the PRD. Want to change the text, or should I take it?

Reply:
Modified the text. WDYT?